lib/lua/mark_eats/ui.lua

```diff
@@ -0,0 +1,309 @@
+--- UI widgets module.
+-- Widgets for paging, tabs, lists, etc.
+--
+-- @module UI
+-- @release v1.0.0
+-- @author Mark Eats
+
+local UI = {}
+UI.__index = UI
+
+
+
+-------- Pages --------
+
+UI.Pages = {}
+UI.Pages.__index = UI.Pages
+
+--- Create a new Pages object.
+-- @param index Selected page, defaults to 1.
+-- @param num_pages Total number of pages, defaults to 3.
+-- @return Instance of Pages.
+function UI.Pages.new(index, num_pages)
+  local pages = {}
```

Comment:
curious if you've considered table constructors? something like:

```lua
local pages = {
  index = index or 1,
  num_pages = num_pages or 3
}
...
```
a little more terse anyway.

more of a win with bigger tables (e.g., `UI.List.new` below).  dunno. food for thought!